x/interchainstaking/types/msgs.go

```diff
@@ -57,23 +63,28 @@ func (msg MsgRequestRedemption) GetSigners() []sdk.AccAddress {
 
 //----------------------------------------------------------------
 
-// IntentsFromString validates and parses the given string into a slice
+// IntentsFromString parses and validates the given string into a slice
 // containing pointers to ValidatorIntent.
 //
 // The combined weights must be 1.0 and the valoper addresses must be valid
-// bech32 strings. (what about zero weights?)
+// bech32 strings.
 //
 // Tokens are comma separated, e.g.
 // "0.3cosmosvaloper1xxxxxxxxx,0.3cosmosvaloper1yyyyyyyyy,0.4cosmosvaloper1zzzzzzzzz".
 func IntentsFromString(input string) ([]*ValidatorIntent, error) {
+	// regexp defining what an intent looks like:
+	// {value}{address}
 	iexpr := regexp.MustCompile(`(\d.\d+)(.+1\w+)`)
```

Comment:
The `.` should be escaped here, else it'll match any character - e.g. this would validate `0L3cosmosvaloper1xxxxxx`

Reply:
good catch